### PR TITLE
Fix Area of Effect Behavior Targets

### PR DESCRIPTION
**Pull request by CodeAX2**
**Description:**
This fixes how entities are targeted during Area of Effect Behaviors. Each AOE Behavior has fields set in the CDClient database to determine who to target. The fields are read in on initialization and are used to filter targets. This fixes issues such as #328 

**Motivation/Context:**
Passive abilities, such as the Samurai's invulnerability at 0 armor, and other in-game effects are handled as an area of effect. The effect targets specific entities based on fields outlined in the CDClient Database. Some behaviors target enemies (such as the buildable sirens in Gnarled Forest), while others target friends (such as the aforementioned Samurai ability). However, there was no logical distinction, and so abilities that should target friends were targeting enemies.

**Types of Changes:**
This is a simple change to the definitions in the area of effect behavior script to read fields from the database related to targeting, as well as changes to destroyable components to allow for validating a target based on these fields.

**Testing:**
This has been tested using multiple players in a variety of situations and behaviors, such as the Sirens and Samurai abilities. Sirens should target enemies (which they do), whereas Samurais should target friends (which they also do)

**Screenshots:**
Below are two screenshots showing the basic functionality.

(1) The siren successfully targets enemies (stromlings) and not friends (players)
Lego Universe Screenshot 2022 01 02 - 14 22 53 95

(2) The Samurai ability does not target enemies (stromlings), but is in effect (via the glowing orange around the player)
Lego Universe Screenshot 2022 01 02 - 14 23 22 60

### CODE REVIEW

**Comment by averysumner** [comment]
Looks good to me, requested review from people who know more about behaviors than me.

**Review by Xiphoseer — APPROVED** [review]
Looks good to me

**Review by Jettford — APPROVED** [review]

File: dGame/dComponents/DestroyableComponent.cpp
Comment (Xiphoseer) [inline dGame/dComponents/DestroyableComponent.cpp:539]: This isn't actually true. There is a separate column in the Factions table that we currently don't use: `friendList`

I'm not certain about the semantics, but I'd suggest you call `DestroyableComponent::IsFriend` as well and we can fix its implementation in a follow up PR.

**Review by Xiphoseer — CHANGES_REQUESTED** [review]
While we're still waiting for a review from @Wincent01 who built this implementation, there was another question raised internally about our `IsFriend` being wrong, which you don't have to fix in this PR but it would be nice to prepare for that at least.

File: dGame/dComponents/DestroyableComponent.cpp
Comment (CodeAX2) [reply dGame/dComponents/DestroyableComponent.cpp]: Okay I see. Originally I just had the one call to IsEnemy since also calling IsFriend takes a bit more time, but since IsFriend is planned to be changed in the future, I have updated the code to determine IsEnemy and IsFriend separately.

**Review by Xiphoseer — APPROVED** [review]

**Comment by IAmMajo** [comment]
I have had this on my server with around 40 players since January 5th and it works perfectly.

**Pull request merged**
